Source/WebKit/WebProcess/WebPage/WebPage.cpp

```diff
@@ -8346,6 +8346,11 @@ void WebPage::requestStorageAccess(RegistrableDomain&& subFrameDomain, Registrab
     });
 }
 
+void WebPage::setLoginStatus(RegistrableDomain&& domain, WebCore::IsLoggedIn loggedInStatus, CompletionHandler<void()>&& completionHandler)
+{
+    WebProcess::singleton().ensureNetworkProcessConnection().connection().sendWithAsyncReply(Messages::NetworkConnectionToWebProcess::SetLoginStatus(WTFMove(domain), loggedInStatus), WTFMove(completionHandler));
```

Comment:
We should use a smart pointer here for `Connection`:
```suggestion
    WebProcess::singleton().ensureNetworkProcessConnection().protectedConnection()->sendWithAsyncReply(Messages::NetworkConnectionToWebProcess::SetLoginStatus(WTFMove(domain), loggedInStatus), WTFMove(completionHandler));
```

Reply:
why do we use protectedConnection() instead of connection?

Reply:
`protectedConnection()` returns a smart pointer instead of a reference, new code is supposed to use smart pointers when calling non-trivial member functions on an object (see https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines#use-smart-pointers-for-object-lifetime-management).